### PR TITLE
Fix quantile gtests coded in namespace cudf::test

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Fixes `cpp/tests/quantiles` gtests source files coded in namespace `cudf::test`
The `tdigest_utilities.cu` was moved to `cpp/tests/utilities` since it is used by quantiles, groupby, reductions tests. Also, the header for the functions defined in this source file is in `cpp/include/cudf_tests/`.

The `cpp/include/cudf_tests/tdigest_utilities.cuh` was also including a source file header from `cudf/tests/groupby` which seemed odd and was corrected by moving the code it needed directly into the `tdigest_utilities.cuh` header. These functions were used by quantiles, groupby, reductions, etc so it made sense for them to be moved into this utility header.

Simple reworking some of the code in `percentile_approx_test.cu` allowed it to become a `.cpp` file as well.
Also made some minor changes to the `tdigest_column_view` class to isolate a functor inside the class instead of the namespace scope.

No function or test has changed just the source code reworked or moved around.

Reference #11734

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/CMakeLists.txt [hunk 1]
```diff
@@ -690,7 +690,7 @@ add_library(cudf::cudf ALIAS cudf)
 add_library(
   cudftestutil STATIC
   tests/io/metadata_utilities.cpp
-  tests/quantiles/tdigest_utilities.cu
```
Comment (bdice) [inline cpp/CMakeLists.txt:693]: Alphabetize this list.

File: cpp/tests/quantiles/quantile_test.cpp [hunk 2]
```diff
@@ -14,26 +14,22 @@
...
-#include <cudf/table/table_view.hpp>
-#include <cudf/types.hpp>
 #include <cudf_test/base_fixture.hpp>
 #include <cudf_test/column_utilities.hpp>
 #include <cudf_test/column_wrapper.hpp>
 #include <cudf_test/type_list_utilities.hpp>
 #include <cudf_test/type_lists.hpp>
+
+#include <cudf/quantiles.hpp>
+//#include <cudf/scalar/scalar.hpp>
```
Comment (bdice) [inline cpp/tests/quantiles/quantile_test.cpp:24]:
```suggestion
```

**Review by bdice — APPROVED** [review]
Approving C++ and CMake with two minor requests.

**Review by ttnghia — APPROVED** [review]

**Comment by davidwendt** [comment]
@gpucibot merge

**Review by vyasr — APPROVED** [review]

**Pull request merged**